compare_hashes/score.h

```diff
@@ -0,0 +1,60 @@
+#ifndef SCORE_H
+#define SCORE_H
+
+#include <cassert>
+
+typedef int location_in_submission;
+typedef unsigned int hash;
+typedef std::string user_id;
+typedef unsigned int version_number;
+
+// represents the plagiarism score for a given submissions, used for the overall rankings file
+class Score {
+public:
+  // CONSTRUCTOR
+  Score(unsigned int hashes_matched, float percent): hashes_matched(hashes_matched), percent(percent), score(-1) {}
+  Score(const Score &other) { copy(other); }

```

Comment:
Is there a reason a copy constructor is being defined? The default copy constructor generated should be fine for this class's member variables

Reply:
Not having it caused a compilation error which Shelly and I were unable to fix without adding the copy constructor.  That was before we figured out the issue with the weird insertion of pairs in an unordered_map so when I have some time tomorrow, I'll take another look at it and see if it can be removed.

Reply:
Oh nevermind, rereading the changes I see that score is being used in the `pair` constructor that's probably the error you were seeing.

You can disregard changing the copy constructor since the copy function is pretty simple or let the compiler define it with `Score(const Score &other) = default;`

Reply:
I'm not sure what was going on but I tried `Score(const Score &other) = default;` and it produced a compiler error similar to the one I get if I remove the copy constructor altogether.  I wish it were possible to get rid of the copy constructor entirely or by doing something like what you suggested but leaving it as-is isn't the end of the world I suppose.